auto-resource-redux/src/FluidBoxScan.lua

```diff
@@ -0,0 +1,215 @@
+            f[filt.name] = old
+          else
+            old.minimum_temperature = math.max(old.minimum_temperature, filt.minimum_temperature)
+            old.maximum_temperature = math.min(old.maximum_temperature, filt.maximum_temperature)
+          end
+          old.output_override = (mining_drill == 'output')
+          old.mining_drill = mining_drill
+          -- correct the max steam temp for generators
+          if max_temp ~= nil and max_temp < old.maximum_temperature then
+            old.maximum_temperature = max_temp
+          end
+          for _, pip in ipairs(pipes) do
+            visited.flows[pip.flow_direction] = true
+          end
+        end
+
+        for ci = 1, #conn do
+          search_fluid_system(conn[ci].owner, sysid, visited, extra_debug)
+        end
+      end
+    end
+  end
+end
+
+--[[
+Determine the best configuration for the entity (requester-tank).
+
+]]
+function FluidBoxScan.autoconfig_request(entity, data)
+  if debug_fluids then
+    log(("autoconfig [%s] '%s' @ %s"):format(entity.unit_number, entity.name, serpent.line(entity.position)))
+  end
+
+  local fluidbox = entity.fluidbox
+  if fluidbox == nil or #fluidbox ~= 1 then
+    return false
+  end
+
+  local sysid = fluidbox.get_fluid_system_id(1)
+  local visited = { filter={}, flows={}, contents={} }
+
+  search_fluid_system(entity, sysid, visited, false)
+
+  if debug_fluids then
+    log(("flows=%s filter=%s"):format(serpent.line(visited.flows), serpent.line(visited.filter)))
+  end
+
+  if next(visited.filter) == nil then
+    if debug_fluids then
+      log("AUTO: Connect to a fluid consumer")
+    end
+    return false
+  end
+
+  -- if there are multitple filters, then we can't auto-config
+  if table_size(visited.filter) ~= 1 then
+    if debug_fluids then
+      log(("AUTO: Too many fluids: %s"):format(serpent.line(visited.filter)))
+    end
+    return false
+  end
+
+  -- if there are multiple flow types, then we can't auto-config
+  if table_size(visited.flows) ~= 1 then
+    if debug_fluids then
+      log("AUTO: Too many connections.")
+    end
+    return false
+  end
+
+  -- if anything is feeding into the fluid network (assembler), then we can't autoconfig
+  if visited.flows.output == true then
+    if debug_fluids then
+      log("AUTO: flows.output=true")
+    end
+    return false
+  end
+
+  -- single input or input-output, find the best fluid temperature
+  local name, filt = next(visited.filter)
+
+  -- if we have an input-output, then we need to wait to see if there is fluid provided
+  if visited.flows['input-output'] == true then
+    if filt.output_override == true then
+      if debug_fluids then
+        log("AUTO: flows.input-output=true output_override=true")
+      end
+      return false
+    end
+  end
+
+  local fluid_proto = game.fluid_prototypes[name]
+  if fluid_proto == nil then
+    return false
+  end
+
+  -- We can auto config!
+  data.fluid    = name
```

Comment:
This function should probably return a new data table (or nil in the case of no data), which the caller would then have to `Util.table_merge` into its own data, instead of trusting that the passed in data object isn't nil.

Reply:
That would work.